transport/zipkin/http.go

```diff
@@ -119,8 +121,11 @@ func (c *HTTPTransport) Flush() (int, error) {
 		return 0, nil
 	}
 	err := c.send(c.batch)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
@yurishkuro I noticed that the jaeger http reporter also drops spans on errors. I don't think dropping spans on errors is a good idea (especially if the error is a 5xx error). 
Should we fix the jaeger http reporter in this commit, or a separate one?